src/Stratis.Bitcoin/P2P/PeerSelector.cs

```diff
@@ -272,6 +272,8 @@ public IEnumerable<PeerAddress> SelectPeersForGetAddrPayload(int peerCount)
             //peers we have not yet connected to and/or that are fresh.
             if (connectedAndHandshaked.Any())
             {
+                connectedAndHandshaked = this.GetPeersNotBannedForBadHandShakes(connectedAndHandshaked).ToList();
```

Comment:
The filtering shouldnt happen here @thefazzer. It should happen in `SelectPreferredPeers`

Reply:
`SelectPeersForGetAddrPayload` is when a peer asks another for it's addresses.

Reply:
Refactored.